pkg/lwip/contrib/sock/udp/lwip_sock_udp.c

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (C) 2016 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ *
+ * @file
+ * @author  Martine Lenders <m.lenders@fu-berlin.de>
+ */
+
+#include <errno.h>
+
+#include "net/ipv4/addr.h"
+#include "net/ipv6/addr.h"
+#include "net/sock/udp.h"
+#include "timex.h"
+
+#include "lwip/api.h"
+#include "lwip/opt.h"
+#include "lwip/sys.h"
+#include "lwip/sock_internal.h"
+
+int sock_udp_create(sock_udp_t *sock, const sock_udp_ep_t *local,
+                    const sock_udp_ep_t *remote, uint16_t flags)
+{
+    assert(sock != NULL);
+    assert(local == NULL || local->port != 0);
+    assert(remote == NULL || remote->port != 0);
+
+    int res;
+    struct netconn *tmp = NULL;
+
+    if ((res = lwip_sock_create(&tmp, (struct _sock_tl_ep *)local,
+                                (struct _sock_tl_ep *)remote, 0, flags,
+                                NETCONN_UDP)) == 0) {
+        sock->conn = tmp;
+    }
+    return res;
+}
+
+void sock_udp_close(sock_udp_t *sock)
+{
+    assert(sock != NULL);
+    if (sock->conn != NULL) {
+        netconn_delete(sock->conn);
+        sock->conn = NULL;
+    }
+}
+
+int sock_udp_get_local(sock_udp_t *sock, sock_udp_ep_t *ep)
+{
+    assert(sock != NULL);
+    return (lwip_sock_get_addr(sock->conn, (struct _sock_tl_ep *)ep,
+                               1)) ? -EADDRNOTAVAIL : 0;
+}
+
+int sock_udp_get_remote(sock_udp_t *sock, sock_udp_ep_t *ep)
+{
+    assert(sock != NULL);
+    return (lwip_sock_get_addr(sock->conn, (struct _sock_tl_ep *)ep,
+                               0)) ? -ENOTCONN : 0;
+}
+
+ssize_t sock_udp_recv(sock_udp_t *sock, void *data, size_t max_len,
+                      uint32_t timeout, sock_udp_ep_t *remote)
+{
```

Comment:
Why `ssize_t`? As far as I can see you're returning `int` and at the end you cast another int to this type...